igvm/commands.py

```diff
@@ -4,14 +4,19 @@
 """
 
 import logging
-from os import environ
+from collections import OrderedDict
 from contextlib import contextmanager, ExitStack
+from ipaddress import ip_address
+from os import environ
 
 from adminapi.dataset import Query
 from adminapi.filters import Any, StartsWith, Contains
 from fabric.colors import green, red, white, yellow
+from fabric.context_managers import settings
+from fabric.decorators import parallel
```

Comment:
Any chance of doing this via standard Python methods? I would prefer to keep Fabric dependencies at minimum.

Reply:
yeah you can probably fiddle with `multiprocessing` or similar.. but no please not :D fabric makes it real easy in this case to parallelize

Reply:
ok after discussion i have implemented a simple parallel executor based on threads without the use of fabric. this also prevents libvirt connections from being established more than once since fabric's `parallel` is based on subprocesses